File: DailyChallenges/CPP/LeetCode_101.cpp
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        function<bool(TreeNode*, TreeNode*)> isSym = [&](TreeNode* a, TreeNode* b){
            return (!a && !b) || (a && b && a->val == b->val && isSym(a->left, b->right) && isSym(a->right, b->left));
        };

        return isSym(root, root);
    }
};

File: DailyChallenges/Scala/LeetCode_101.scala
object Solution {
    def isSymmetric(root: TreeNode): Boolean = {
        def helper(a: TreeNode, b: TreeNode): Boolean = 
            (a==null && b==null) || 
            (a != null && b!= null && a.value == b.value && 
            helper(a.left, b.right) && helper(a.right, b.left))

        helper(root, root)
    }
};
